In [1]:
import pandas as pd
import numpy as np

import matplotlib as mpl
import matplotlib.pyplot as plt
import seaborn as sns

import warnings
warnings.filterwarnings('ignore')

pd.set_option('display.max_columns', None)

mpl.rcParams['axes.unicode_minus'] = False
plt.rcParams['font.family'] = 'Malgun Gothic'


pd.set_option('display.max_rows', 500)
pd.set_option('display.max_columns', 500)

In [2]:
df = pd.read_csv('./1년 1시간별 Ls, Ws.csv', encoding = 'cp949')
df

자치구           대여소명  대여소번호     년  월  시간    lambda        mu    c  \
0       중랑구  (구)신한은행 중랑교지점   1442  2020  6   0  3.800000  2.700000  6.0   
1       중랑구  (구)신한은행 중랑교지점   1442  2020  6   1  3.700000  2.133333  6.0   
2       중랑구  (구)신한은행 중랑교지점   1442  2020  6   2  2.133333  1.266667  6.0   
3       중랑구  (구)신한은행 중랑교지점   1442  2020  6   3  1.566667  0.833333  6.0   
4       중랑구  (구)신한은행 중랑교지점   1442  2020  6   4  1.333333  0.533333  6.0   
...     ...            ...    ...   ... ..  ..       ...       ...  ...   
694939  강서구        힐스테이트에코   2720  2021  5  19  7.933333  8.066667  9.0   
694940  강서구        힐스테이트에코   2720  2021  5  20  5.166667  7.800000  9.0   
694941  강서구        힐스테이트에코   2720  2021  5  21  3.400000  6.933333  9.0   
694942  강서구        힐스테이트에코   2720  2021  5  22  1.800000  3.000000  9.0   
694943  강서구        힐스테이트에코   2720  2021  5  23  1.233333  1.166667  9.0   

             rho       p_0       p_1      p_1+       p_c       L_s        W_s  
0       1.407407  0.040995  0.057696  0.959005  0.318602  4.249820   1.641292  
1       1.734375  0.015893  0.027565  0.984107  0.432587  4.789792   2.281475  
2       1.684211  0.018275  0.030779  0.981725  0.417101  4.725431   3.800050  
3       1.880000  0.010731  0.020174  0.989269  0.473793  4.948997   6.003217  
4       2.500000  0.002462  0.006154  0.997538  0.600985  5.344821  10.046270  
...          ...       ...       ...       ...       ...       ...        ...  
694939  0.983471  0.107665  0.105886  0.892335  0.092668  4.362561   0.606066  
694940  0.662393  0.343188  0.227325  0.656812  0.008425  1.796723   0.350708  
694941  0.490385  0.510026  0.250109  0.489974  0.000836  0.954216   0.280887  
694942  0.600000  0.402433  0.241460  0.597567  0.004056  1.439166   0.802792  
694943  1.057143  0.076892  0.081286  0.923108  0.126790  4.956086   4.601924  

[694944 rows x 16 columns]

In [3]:
df = df[df['대여소명'] == '한성백제역 2번 출구']
df

자치구         대여소명  대여소번호     년   월  시간     lambda         mu     c  \
655776  송파구  한성백제역 2번 출구   2621  2020   6   0   4.500000   3.066667  16.0   
655777  송파구  한성백제역 2번 출구   2621  2020   6   1   2.966667   2.133333  16.0   
655778  송파구  한성백제역 2번 출구   2621  2020   6   2   1.800000   2.100000  16.0   
655779  송파구  한성백제역 2번 출구   2621  2020   6   3   1.233333   1.833333  16.0   
655780  송파구  한성백제역 2번 출구   2621  2020   6   4   0.800000   0.566667  16.0   
655781  송파구  한성백제역 2번 출구   2621  2020   6   5   1.000000   0.700000  16.0   
655782  송파구  한성백제역 2번 출구   2621  2020   6   6   2.200000   1.800000  16.0   
655783  송파구  한성백제역 2번 출구   2621  2020   6   7   2.400000   4.700000  16.0   
655784  송파구  한성백제역 2번 출구   2621  2020   6   8   8.766667   6.633333  16.0   
655785  송파구  한성백제역 2번 출구   2621  2020   6   9   4.700000   5.200000  16.0   
655786  송파구  한성백제역 2번 출구   2621  2020   6  10   4.666667   4.433333  16.0   
655787  송파구  한성백제역 2번 출구   2621  2020   6  11   5.100000   4.566667  16.0   
655788  송파구  한성백제역 2번 출구   2621  2020   6  12   5.766667   5.266667  16.0   
655789  송파구  한성백제역 2번 출구   2621  2020   6  13   6.266667   5.800000  16.0   
655790  송파구  한성백제역 2번 출구   2621  2020   6  14   5.800000   6.700000  16.0   
655791  송파구  한성백제역 2번 출구   2621  2020   6  15   8.266667   8.766667  16.0   
655792  송파구  한성백제역 2번 출구   2621  2020   6  16   8.900000  10.866667  16.0   
655793  송파구  한성백제역 2번 출구   2621  2020   6  17  12.166667  12.866667  16.0   
655794  송파구  한성백제역 2번 출구   2621  2020   6  18  16.566667  16.700000  16.0   
655795  송파구  한성백제역 2번 출구   2621  2020   6  19  15.600000  15.533333  16.0   
655796  송파구  한성백제역 2번 출구   2621  2020   6  20  16.700000  16.533333  16.0   
655797  송파구  한성백제역 2번 출구   2621  2020   6  21  14.700000  15.766667  16.0   
655798  송파구  한성백제역 2번 출구   2621  2020   6  22  14.066667  12.333333  16.0   
655799  송파구  한성백제역 2번 출구   2621  2020   6  23   7.400000   7.933333  16.0   
655800  송파구  한성백제역 2번 출구   2621  2020   7   0   4.266667   3.933333  16.0   
655801  송파구  한성백제역 2번 출구   2621  2020   7   1   3.300000   2.400000  16.0   
655802  송파구  한성백제역 2번 출구   2621  2020   7   2   1.933333   2.333333  16.0   
655803  송파구  한성백제역 2번 출구   2621  2020   7   3   1.133333   1.700000  16.0   
655804  송파구  한성백제역 2번 출구   2621  2020   7   4   0.833333   0.800000  16.0   
655805  송파구  한성백제역 2번 출구   2621  2020   7   5   1.066667   0.600000  16.0   
655806  송파구  한성백제역 2번 출구   2621  2020   7   6   1.233333   1.533333  16.0   
655807  송파구  한성백제역 2번 출구   2621  2020   7   7   1.100000   2.300000  16.0   
655808  송파구  한성백제역 2번 출구   2621  2020   7   8   6.300000   4.566667  16.0   
655809  송파구  한성백제역 2번 출구   2621  2020   7   9   4.600000   3.233333  16.0   
655810  송파구  한성백제역 2번 출구   2621  2020   7  10   4.900000   4.733333  16.0   
655811  송파구  한성백제역 2번 출구   2621  2020   7  11   3.733333   5.033333  16.0   
655812  송파구  한성백제역 2번 출구   2621  2020   7  12   5.300000   4.900000  16.0   
655813  송파구  한성백제역 2번 출구   2621  2020   7  13   5.500000   4.533333  16.0   
655814  송파구  한성백제역 2번 출구   2621  2020   7  14   6.233333   6.333333  16.0   
655815  송파구  한성백제역 2번 출구   2621  2020   7  15   8.066667   7.400000  16.0   
655816  송파구  한성백제역 2번 출구   2621  2020   7  16   9.066667  10.033333  16.0   
655817  송파구  한성백제역 2번 출구   2621  2020   7  17  11.566667  12.133333  16.0   
655818  송파구  한성백제역 2번 출구   2621  2020   7  18  13.333333  14.733333  16.0   
655819  송파구  한성백제역 2번 출구   2621  2020   7  19  13.866667  13.133333  16.0   
655820  송파구  한성백제역 2번 출구   2621  2020   7  20  15.133333  15.233333  16.0   
655821  송파구  한성백제역 2번 출구   2621  2020   7  21  14.500000  14.600000  16.0   
655822  송파구  한성백제역 2번 출구   2621  2020   7  22  11.300000  11.400000  16.0   
655823  송파구  한성백제역 2번 출구   2621  2020   7  23   6.733333   5.733333  16.0   
655824  송파구  한성백제역 2번 출구   2621  2020   8   0   2.433333   2.333333  16.0   
655825  송파구  한성백제역 2번 출구   2621  2020   8   1   1.166667   1.100000  16.0   
655826  송파구  한성백제역 2번 출구   2621  2020   8   2   1.700000   1.566667  16.0   
6558

In [4]:
df = df.drop(['대여소번호', 'c', 'rho', 'p_c', 'p_0', 'p_1'], axis = 1)
df

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
655776  송파구  한성백제역 2번 출구  2020   6   0   4.500000   3.066667  0.999310   
655777  송파구  한성백제역 2번 출구  2020   6   1   2.966667   2.133333  0.998559   
655778  송파구  한성백제역 2번 출구  2020   6   2   1.800000   2.100000  0.845933   
655779  송파구  한성백제역 2번 출구  2020   6   3   1.233333   1.833333  0.672339   
655780  송파구  한성백제역 2번 출구  2020   6   4   0.800000   0.566667  0.998825   
655781  송파구  한성백제역 2번 출구  2020   6   5   1.000000   0.700000  0.999001   
655782  송파구  한성백제역 2번 출구  2020   6   6   2.200000   1.800000  0.992418   
655783  송파구  한성백제역 2번 출구  2020   6   7   2.400000   4.700000  0.510633   
655784  송파구  한성백제역 2번 출구  2020   6   8   8.766667   6.633333  0.997166   
655785  송파구  한성백제역 2번 출구  2020   6   9   4.700000   5.200000  0.882838   
655786  송파구  한성백제역 2번 출구  2020   6  10   4.666667   4.433333  0.962181   
655787  송파구  한성백제역 2번 출구  2020   6  11   5.100000   4.566667  0.978915   
655788  송파구  한성백제역 2번 출구  2020   6  12   5.766667   5.266667  0.974154   
655789  송파구  한성백제역 2번 출구  2020   6  13   6.266667   5.800000  0.970494   
655790  송파구  한성백제역 2번 출구  2020   6  14   5.800000   6.700000  0.853016   
655791  송파구  한성백제역 2번 출구  2020   6  15   8.266667   8.766667  0.909685   
655792  송파구  한성백제역 2번 출구  2020   6  16   8.900000  10.866667  0.812731   
655793  송파구  한성백제역 2번 출구  2020   6  17  12.166667  12.866667  0.911342   
655794  송파구  한성백제역 2번 출구  2020   6  18  16.566667  16.700000  0.937329   
655795  송파구  한성백제역 2번 출구  2020   6  19  15.600000  15.533333  0.943170   
655796  송파구  한성백제역 2번 출구  2020   6  20  16.700000  16.533333  0.945778   
655797  송파구  한성백제역 2번 출구  2020   6  21  14.700000  15.766667  0.902803   
655798  송파구  한성백제역 2번 출구  2020   6  22  14.066667  12.333333  0.983172   
655799  송파구  한성백제역 2번 출구  2020   6  23   7.400000   7.933333  0.903085   
655800  송파구  한성백제역 2번 출구  2020   7   0   4.266667   3.933333  0.971622   
655801  송파구  한성백제역 2번 출구  2020   7   1   3.300000   2.400000  0.998322   
655802  송파구  한성백제역 2번 출구  2020   7   2   1.933333   2.333333  0.821263   
655803  송파구  한성백제역 2번 출구  2020   7   3   1.133333   1.700000  0.666328   
655804  송파구  한성백제역 2번 출구  2020   7   4   0.833333   0.800000  0.958402   
655805  송파구  한성백제역 2번 출구  2020   7   5   1.066667   0.600000  0.999956   
655806  송파구  한성백제역 2번 출구  2020   7   6   1.233333   1.533333  0.799395   
655807  송파구  한성백제역 2번 출구  2020   7   7   1.100000   2.300000  0.478259   
655808  송파구  한성백제역 2번 출구  2020   7   8   6.300000   4.566667  0.998395   
655809  송파구  한성백제역 2번 출구  2020   7   9   4.600000   3.233333  0.998943   
655810  송파구  한성백제역 2번 출구  2020   7  10   4.900000   4.733333  0.956036   
655811  송파구  한성백제역 2번 출구  2020   7  11   3.733333   5.033333  0.740104   
655812  송파구  한성백제역 2번 출구  2020   7  12   5.300000   4.900000  0.970807   
655813  송파구  한성백제역 2번 출구  2020   7  13   5.500000   4.533333  0.991714   
655814  송파구  한성백제역 2번 출구  2020   7  14   6.233333   6.333333  0.933392   
655815  송파구  한성백제역 2번 출구  2020   7  15   8.066667   7.400000  0.972976   
655816  송파구  한성백제역 2번 출구  2020   7  16   9.066667  10.033333  0.882696   
655817  송파구  한성백제역 2번 출구  2020   7  17  11.566667  12.133333  0.916079   
655818  송파구  한성백제역 2번 출구  2020   7  18  13.333333  14.733333  0.883670   
655819  송파구  한성백제역 2번 출구  2020   7  19  13.866667  13.133333  0.963230   
655820  송파구  한성백제역 2번 출구  2020   7  20  15.133333  15.233333  0.938026   
655821  송파구  한성백제역 2번 출구  2020   7  21  14.500000  14.600000  0.937887   
655822  송파구  한성백제역 2번 출구  2020   7  22  11.300000  11.400000  0.936939   
655823  송파구  한성백제역 2번 출구  2020   7  23   6.733333   5.733333  0.987872   
655824  송파구  한성백제역 2번 출구  2020   8   0   2.433333   2.333333  0.958827   
655825  송파구  한성백제역 2번 출구  2020   8   1   1.166667   1.100000  0.964745   
655826  송파구  한성백제역 2번 출구  2020   8   2   1.700000   1.566667  0.971715   
655827  송파구  한성백제역 2번 출구  2020   8   3   1.300000   1.733333  0.748107   
655828  송파구  한성백제역 2번 출구  2020   8   4   0.800000   0.666667  0.990560   
655829  송파구 

In [5]:
data = pd.read_csv('./한성백제역 2번 출구 균형, 불균형.csv', encoding = 'cp949')
data

자치구         대여소명     년   월  시간        재고량      3사분위수     2사분위수     1사분위수  \
0    송파구  한성백제역 2번 출구  2020   6   0   4.066667  12.066667  7.633333  5.566667   
1    송파구  한성백제역 2번 출구  2020   6   1   4.966667  12.066667  7.633333  5.566667   
2    송파구  한성백제역 2번 출구  2020   6   2   4.900000  12.066667  7.633333  5.566667   
3    송파구  한성백제역 2번 출구  2020   6   3   4.800000  12.066667  7.633333  5.566667   
4    송파구  한성백제역 2번 출구  2020   6   4   5.066667  12.066667  7.633333  5.566667   
5    송파구  한성백제역 2번 출구  2020   6   5   5.433333  12.066667  7.633333  5.566667   
6    송파구  한성백제역 2번 출구  2020   6   6   5.933333  12.066667  7.633333  5.566667   
7    송파구  한성백제역 2번 출구  2020   6   7   3.866667  12.066667  7.633333  5.566667   
8    송파구  한성백제역 2번 출구  2020   6   8   5.166667  12.066667  7.633333  5.566667   
9    송파구  한성백제역 2번 출구  2020   6   9   4.666667  12.066667  7.633333  5.566667   
10   송파구  한성백제역 2번 출구  2020   6  10   4.966667  12.066667  7.633333  5.566667   
11   송파구  한성백제역 2번 출구  2020   6  11   5.600000  12.066667  7.633333  5.566667   
12   송파구  한성백제역 2번 출구  2020   6  12   6.100000  12.066667  7.633333  5.566667   
13   송파구  한성백제역 2번 출구  2020   6  13   6.600000  12.066667  7.633333  5.566667   
14   송파구  한성백제역 2번 출구  2020   6  14   5.700000  12.066667  7.633333  5.566667   
15   송파구  한성백제역 2번 출구  2020   6  15   5.200000  12.066667  7.633333  5.566667   
16   송파구  한성백제역 2번 출구  2020   6  16   3.233333  12.066667  7.633333  5.566667   
17   송파구  한성백제역 2번 출구  2020   6  17   2.533333  12.066667  7.633333  5.566667   
18   송파구  한성백제역 2번 출구  2020   6  18   2.400000  12.066667  7.633333  5.566667   
19   송파구  한성백제역 2번 출구  2020   6  19   2.466667  12.066667  7.633333  5.566667   
20   송파구  한성백제역 2번 출구  2020   6  20   2.666667  12.066667  7.633333  5.566667   
21   송파구  한성백제역 2번 출구  2020   6  21   1.633333  12.066667  7.633333  5.566667   
22   송파구  한성백제역 2번 출구  2020   6  22   3.366667  12.066667  7.633333  5.566667   
23   송파구  한성백제역 2번 출구  2020   6  23   2.900000  12.066667  7.633333  5.566667   
24   송파구  한성백제역 2번 출구  2020   7   0   5.700000  12.066667  7.633333  5.566667   
25   송파구  한성백제역 2번 출구  2020   7   1   6.666667  12.066667  7.633333  5.566667   
26   송파구  한성백제역 2번 출구  2020   7   2   6.600000  12.066667  7.633333  5.566667   
27   송파구  한성백제역 2번 출구  2020   7   3   6.566667  12.066667  7.633333  5.566667   
28   송파구  한성백제역 2번 출구  2020   7   4   6.966667  12.066667  7.633333  5.566667   
29   송파구  한성백제역 2번 출구  2020   7   5   7.466667  12.066667  7.633333  5.566667   
30   송파구  한성백제역 2번 출구  2020   7   6   7.300000  12.066667  7.633333  5.566667   
31   송파구  한성백제역 2번 출구  2020   7   7   6.633333  12.066667  7.633333  5.566667   
32   송파구  한성백제역 2번 출구  2020   7   8   3.400000  12.066667  7.633333  5.566667   
33   송파구  한성백제역 2번 출구  2020   7   9   4.800000  12.066667  7.633333  5.566667   
34   송파구  한성백제역 2번 출구  2020   7  10   5.066667  12.066667  7.633333  5.566667   
35   송파구  한성백제역 2번 출구  2020   7  11   4.066667  12.066667  7.633333  5.566667   
36   송파구  한성백제역 2번 출구  2020   7  12   4.500000  12.066667  7.633333  5.566667   
37   송파구  한성백제역 2번 출구  2020   7  13   5.500000  12.066667  7.633333  5.566667   
38   송파구  한성백제역 2번 출구  2020   7  14   5.400000  12.066667  7.633333  5.566667   
39   송파구  한성백제역 2번 출구  2020   7  15   6.166667  12.066667  7.633333  5.566667   
40   송파구  한성백제역 2번 출구  2020   7  16   5.200000  12.066667  7.633333  5.566667   
41   송파구  한성백제역 2번 출구  2020   7  17   4.633333  12.066667  7.633333  5.566667   
42   송파구  한성백제역 2번 출구  2020   7  18   3.300000  12.066667  7.633333  5.566667   
43   송파구  한성백제역 2번 출구  2020   7  19   4.166667  12.066667  7.633333  5.566667   
44   송파구  한성백제역 2번 출구  2020   7  20   4.066667  12.066667  7.633333  5.566667   
45   송파구  한성백제역 2번 출구  2020   7  21   3.966667  12.066667  7.633333  5.566667   
46   송파구  한성백제역 2번 출구  2020   7  22   3.966667  12.066667  7.633333  5.566667   
47   송파구  한성백제역 2번 출구  2020   7  23   5.033333  12.066667  7.633333  5.566667   
48   송파구  한성백제역 2번 출구  2020   8   0 

In [6]:
df_new = pd.merge(df,data, on = ['자치구', '대여소명','년', '월', '시간'],how='outer')
df_new

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  한성백제역 2번 출구  2020   6   0   4.500000   3.066667  0.999310   
1    송파구  한성백제역 2번 출구  2020   6   1   2.966667   2.133333  0.998559   
2    송파구  한성백제역 2번 출구  2020   6   2   1.800000   2.100000  0.845933   
3    송파구  한성백제역 2번 출구  2020   6   3   1.233333   1.833333  0.672339   
4    송파구  한성백제역 2번 출구  2020   6   4   0.800000   0.566667  0.998825   
5    송파구  한성백제역 2번 출구  2020   6   5   1.000000   0.700000  0.999001   
6    송파구  한성백제역 2번 출구  2020   6   6   2.200000   1.800000  0.992418   
7    송파구  한성백제역 2번 출구  2020   6   7   2.400000   4.700000  0.510633   
8    송파구  한성백제역 2번 출구  2020   6   8   8.766667   6.633333  0.997166   
9    송파구  한성백제역 2번 출구  2020   6   9   4.700000   5.200000  0.882838   
10   송파구  한성백제역 2번 출구  2020   6  10   4.666667   4.433333  0.962181   
11   송파구  한성백제역 2번 출구  2020   6  11   5.100000   4.566667  0.978915   
12   송파구  한성백제역 2번 출구  2020   6  12   5.766667   5.266667  0.974154   
13   송파구  한성백제역 2번 출구  2020   6  13   6.266667   5.800000  0.970494   
14   송파구  한성백제역 2번 출구  2020   6  14   5.800000   6.700000  0.853016   
15   송파구  한성백제역 2번 출구  2020   6  15   8.266667   8.766667  0.909685   
16   송파구  한성백제역 2번 출구  2020   6  16   8.900000  10.866667  0.812731   
17   송파구  한성백제역 2번 출구  2020   6  17  12.166667  12.866667  0.911342   
18   송파구  한성백제역 2번 출구  2020   6  18  16.566667  16.700000  0.937329   
19   송파구  한성백제역 2번 출구  2020   6  19  15.600000  15.533333  0.943170   
20   송파구  한성백제역 2번 출구  2020   6  20  16.700000  16.533333  0.945778   
21   송파구  한성백제역 2번 출구  2020   6  21  14.700000  15.766667  0.902803   
22   송파구  한성백제역 2번 출구  2020   6  22  14.066667  12.333333  0.983172   
23   송파구  한성백제역 2번 출구  2020   6  23   7.400000   7.933333  0.903085   
24   송파구  한성백제역 2번 출구  2020   7   0   4.266667   3.933333  0.971622   
25   송파구  한성백제역 2번 출구  2020   7   1   3.300000   2.400000  0.998322   
26   송파구  한성백제역 2번 출구  2020   7   2   1.933333   2.333333  0.821263   
27   송파구  한성백제역 2번 출구  2020   7   3   1.133333   1.700000  0.666328   
28   송파구  한성백제역 2번 출구  2020   7   4   0.833333   0.800000  0.958402   
29   송파구  한성백제역 2번 출구  2020   7   5   1.066667   0.600000  0.999956   
30   송파구  한성백제역 2번 출구  2020   7   6   1.233333   1.533333  0.799395   
31   송파구  한성백제역 2번 출구  2020   7   7   1.100000   2.300000  0.478259   
32   송파구  한성백제역 2번 출구  2020   7   8   6.300000   4.566667  0.998395   
33   송파구  한성백제역 2번 출구  2020   7   9   4.600000   3.233333  0.998943   
34   송파구  한성백제역 2번 출구  2020   7  10   4.900000   4.733333  0.956036   
35   송파구  한성백제역 2번 출구  2020   7  11   3.733333   5.033333  0.740104   
36   송파구  한성백제역 2번 출구  2020   7  12   5.300000   4.900000  0.970807   
37   송파구  한성백제역 2번 출구  2020   7  13   5.500000   4.533333  0.991714   
38   송파구  한성백제역 2번 출구  2020   7  14   6.233333   6.333333  0.933392   
39   송파구  한성백제역 2번 출구  2020   7  15   8.066667   7.400000  0.972976   
40   송파구  한성백제역 2번 출구  2020   7  16   9.066667  10.033333  0.882696   
41   송파구  한성백제역 2번 출구  2020   7  17  11.566667  12.133333  0.916079   
42   송파구  한성백제역 2번 출구  2020   7  18  13.333333  14.733333  0.883670   
43   송파구  한성백제역 2번 출구  2020   7  19  13.866667  13.133333  0.963230   
44   송파구  한성백제역 2번 출구  2020   7  20  15.133333  15.233333  0.938026   
45   송파구  한성백제역 2번 출구  2020   7  21  14.500000  14.600000  0.937887   
46   송파구  한성백제역 2번 출구  2020   7  22  11.300000  11.400000  0.936939   
47   송파구  한성백제역 2번 출구  2020   7  23   6.733333   5.733333  0.987872   
48   송파구  한성백제역 2번 출구  2020   8   0   2.433333   2.333333  0.958827   
49   송파구  한성백제역 2번 출구  2020   8   1   1.166667   1.100000  0.964745   
50   송파구  한성백제역 2번 출구  2020   8   2   1.700000   1.566667  0.971715   
51   송파구  한성백제역 2번 출구  2020   8   3   1.300000   1.733333  0.748107   
52   송파구  한성백제역 2번 출구  2020   8   4   0.800000   0.666667  0.990560   
53   송파구  한성백제역 2번 출구  2020   8   5   0.800000   0.566667  0.998825   
54   송파구  한성백제역 2번 출구  2020   8   6   0.733333   0.700000  0.960490   
55   송파구  한성백제역 2번 출구  2020  

In [7]:
df_new['내 균형'] = '균형'
df_new

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  한성백제역 2번 출구  2020   6   0   4.500000   3.066667  0.999310   
1    송파구  한성백제역 2번 출구  2020   6   1   2.966667   2.133333  0.998559   
2    송파구  한성백제역 2번 출구  2020   6   2   1.800000   2.100000  0.845933   
3    송파구  한성백제역 2번 출구  2020   6   3   1.233333   1.833333  0.672339   
4    송파구  한성백제역 2번 출구  2020   6   4   0.800000   0.566667  0.998825   
5    송파구  한성백제역 2번 출구  2020   6   5   1.000000   0.700000  0.999001   
6    송파구  한성백제역 2번 출구  2020   6   6   2.200000   1.800000  0.992418   
7    송파구  한성백제역 2번 출구  2020   6   7   2.400000   4.700000  0.510633   
8    송파구  한성백제역 2번 출구  2020   6   8   8.766667   6.633333  0.997166   
9    송파구  한성백제역 2번 출구  2020   6   9   4.700000   5.200000  0.882838   
10   송파구  한성백제역 2번 출구  2020   6  10   4.666667   4.433333  0.962181   
11   송파구  한성백제역 2번 출구  2020   6  11   5.100000   4.566667  0.978915   
12   송파구  한성백제역 2번 출구  2020   6  12   5.766667   5.266667  0.974154   
13   송파구  한성백제역 2번 출구  2020   6  13   6.266667   5.800000  0.970494   
14   송파구  한성백제역 2번 출구  2020   6  14   5.800000   6.700000  0.853016   
15   송파구  한성백제역 2번 출구  2020   6  15   8.266667   8.766667  0.909685   
16   송파구  한성백제역 2번 출구  2020   6  16   8.900000  10.866667  0.812731   
17   송파구  한성백제역 2번 출구  2020   6  17  12.166667  12.866667  0.911342   
18   송파구  한성백제역 2번 출구  2020   6  18  16.566667  16.700000  0.937329   
19   송파구  한성백제역 2번 출구  2020   6  19  15.600000  15.533333  0.943170   
20   송파구  한성백제역 2번 출구  2020   6  20  16.700000  16.533333  0.945778   
21   송파구  한성백제역 2번 출구  2020   6  21  14.700000  15.766667  0.902803   
22   송파구  한성백제역 2번 출구  2020   6  22  14.066667  12.333333  0.983172   
23   송파구  한성백제역 2번 출구  2020   6  23   7.400000   7.933333  0.903085   
24   송파구  한성백제역 2번 출구  2020   7   0   4.266667   3.933333  0.971622   
25   송파구  한성백제역 2번 출구  2020   7   1   3.300000   2.400000  0.998322   
26   송파구  한성백제역 2번 출구  2020   7   2   1.933333   2.333333  0.821263   
27   송파구  한성백제역 2번 출구  2020   7   3   1.133333   1.700000  0.666328   
28   송파구  한성백제역 2번 출구  2020   7   4   0.833333   0.800000  0.958402   
29   송파구  한성백제역 2번 출구  2020   7   5   1.066667   0.600000  0.999956   
30   송파구  한성백제역 2번 출구  2020   7   6   1.233333   1.533333  0.799395   
31   송파구  한성백제역 2번 출구  2020   7   7   1.100000   2.300000  0.478259   
32   송파구  한성백제역 2번 출구  2020   7   8   6.300000   4.566667  0.998395   
33   송파구  한성백제역 2번 출구  2020   7   9   4.600000   3.233333  0.998943   
34   송파구  한성백제역 2번 출구  2020   7  10   4.900000   4.733333  0.956036   
35   송파구  한성백제역 2번 출구  2020   7  11   3.733333   5.033333  0.740104   
36   송파구  한성백제역 2번 출구  2020   7  12   5.300000   4.900000  0.970807   
37   송파구  한성백제역 2번 출구  2020   7  13   5.500000   4.533333  0.991714   
38   송파구  한성백제역 2번 출구  2020   7  14   6.233333   6.333333  0.933392   
39   송파구  한성백제역 2번 출구  2020   7  15   8.066667   7.400000  0.972976   
40   송파구  한성백제역 2번 출구  2020   7  16   9.066667  10.033333  0.882696   
41   송파구  한성백제역 2번 출구  2020   7  17  11.566667  12.133333  0.916079   
42   송파구  한성백제역 2번 출구  2020   7  18  13.333333  14.733333  0.883670   
43   송파구  한성백제역 2번 출구  2020   7  19  13.866667  13.133333  0.963230   
44   송파구  한성백제역 2번 출구  2020   7  20  15.133333  15.233333  0.938026   
45   송파구  한성백제역 2번 출구  2020   7  21  14.500000  14.600000  0.937887   
46   송파구  한성백제역 2번 출구  2020   7  22  11.300000  11.400000  0.936939   
47   송파구  한성백제역 2번 출구  2020   7  23   6.733333   5.733333  0.987872   
48   송파구  한성백제역 2번 출구  2020   8   0   2.433333   2.333333  0.958827   
49   송파구  한성백제역 2번 출구  2020   8   1   1.166667   1.100000  0.964745   
50   송파구  한성백제역 2번 출구  2020   8   2   1.700000   1.566667  0.971715   
51   송파구  한성백제역 2번 출구  2020   8   3   1.300000   1.733333  0.748107   
52   송파구  한성백제역 2번 출구  2020   8   4   0.800000   0.666667  0.990560   
53   송파구  한성백제역 2번 출구  2020   8   5   0.800000   0.566667  0.998825   
54   송파구  한성백제역 2번 출구  2020   8   6   0.733333   0.700000  0.960490   
55   송파구  한성백제역 2번 출구  2020  

# 일단 L_s, 기존균형 비교

## L_s가 Q1보다 작거나, Q3보다 크면 불균형

### L_s가 q1보다 작지만 이때 p_1+이 충분히 크다면 균형
- ls가 q1보다 작고 p_1+가 0.5보다 작으면 불균형

In [8]:
# ls가 q1보다 작으므로 불균형

e = df_new[(df_new['L_s'] < df_new['1사분위수']) & (df_new['p_1+'] < 0.6)]
e

자치구         대여소명     년   월  시간    lambda        mu      p_1+       L_s  \
7    송파구  한성백제역 2번 출구  2020   6   7  2.400000  4.700000  0.510633  1.043293   
31   송파구  한성백제역 2번 출구  2020   7   7  1.100000  2.300000  0.478259  0.916606   
55   송파구  한성백제역 2번 출구  2020   8   7  0.666667  1.400000  0.476189  0.909034   
79   송파구  한성백제역 2번 출구  2020   9   7  1.900000  3.633333  0.522928  1.095876   
103  송파구  한성백제역 2번 출구  2020  10   7  1.200000  3.066667  0.391304  0.642855   
127  송파구  한성백제역 2번 출구  2020  11   7  1.200000  2.233333  0.537301  1.160849   
146  송파구  한성백제역 2번 출구  2020  12   2  0.300000  1.066667  0.281250  0.391304   
170  송파구  한성백제역 2번 출구  2021   1   2  0.100000  0.200000  0.499996  0.999870   
172  송파구  한성백제역 2번 출구  2021   1   4  0.000000  0.000000  0.000000  0.000000   
243  송파구  한성백제역 2번 출구  2021   4   3  0.200000  0.400000  0.499996  0.999870   
246  송파구  한성백제역 2번 출구  2021   4   6  0.600000  1.366667  0.439024  0.782594   
247  송파구  한성백제역 2번 출구  2021   4   7  1.866667  3.366667  0.554436  1.243692   
270  송파구  한성백제역 2번 출구  2021   5   6  0.733333  1.800000  0.407407  0.687496   

          W_s        재고량      3사분위수     2사분위수     1사분위수   균형 내 균형  
7    0.434710   3.866667  12.066667  7.633333  5.566667  불균형   균형  
31   0.833281   6.633333  12.066667  7.633333  5.566667   균형   균형  
55   1.363556   6.766667  12.066667  7.633333  5.566667   균형   균형  
79   0.576785   9.666667  12.066667  7.633333  5.566667   균형   균형  
103  0.535713  12.133333  12.066667  7.633333  5.566667  불균형   균형  
127  0.967396  13.566667  12.066667  7.633333  5.566667  불균형   균형  
146  1.304348   6.766667  12.066667  7.633333  5.566667   균형   균형  
170  9.998779   9.266667  12.066667  7.633333  5.566667   균형   균형  
172  0.000000   9.466667  12.066667  7.633333  5.566667   균형   균형  
243  4.999390   7.400000  12.066667  7.633333  5.566667   균형   균형  
246  1.304326   8.133333  12.066667  7.633333  5.566667   균형   균형  
247  0.666288   6.933333  12.066667  7.633333  5.566667   균형   균형  
270  0.937495  15.700000  12.066667  7.633333  5.566667  불균형   균형

In [9]:
for i in range(len(df_new)):
    if (df_new['L_s'][i] < df_new['1사분위수'][i]) & (df_new['p_1+'][i] < 0.6):
        df_new['내 균형'][i] = '불균형'

In [10]:
# ls가 q3보다 크므로 불균형

r = df_new[(df_new['L_s'] > df_new['3사분위수']) & (df_new['L_s'] > (15*0.7))]
r

자치구         대여소명     년   월  시간     lambda        mu      p_1+        L_s  \
0    송파구  한성백제역 2번 출구  2020   6   0   4.500000  3.066667  0.999310  13.885573   
1    송파구  한성백제역 2번 출구  2020   6   1   2.966667  2.133333  0.998559  13.502730   
4    송파구  한성백제역 2번 출구  2020   6   4   0.800000  0.566667  0.998825  13.619927   
5    송파구  한성백제역 2번 출구  2020   6   5   1.000000  0.700000  0.999001  13.706306   
6    송파구  한성백제역 2번 출구  2020   6   6   2.200000  1.800000  0.992418  12.080052   
8    송파구  한성백제역 2번 출구  2020   6   8   8.766667  6.633333  0.997166  13.040426   
25   송파구  한성백제역 2번 출구  2020   7   1   3.300000  2.400000  0.998322  13.409408   
29   송파구  한성백제역 2번 출구  2020   7   5   1.066667  0.600000  0.999956  14.715246   
32   송파구  한성백제역 2번 출구  2020   7   8   6.300000  4.566667  0.998395  13.437277   
33   송파구  한성백제역 2번 출구  2020   7   9   4.600000  3.233333  0.998943  13.676678   
53   송파구  한성백제역 2번 출구  2020   8   5   0.800000  0.566667  0.998825  13.619927   
59   송파구  한성백제역 2번 출구  2020   8  11   3.433333  2.466667  0.998576  13.510039   
77   송파구  한성백제역 2번 출구  2020   9   5   0.800000  0.200000  1.000000  15.666667   
95   송파구  한성백제역 2번 출구  2020   9  23   6.466667  5.133333  0.994771  12.492238   
98   송파구  한성백제역 2번 출구  2020  10   2   1.966667  1.300000  0.999549  14.064943   
100  송파구  한성백제역 2번 출구  2020  10   4   0.733333  0.500000  0.999305  13.882463   
101  송파구  한성백제역 2번 출구  2020  10   5   0.633333  0.300000  0.999997  15.100052   
124  송파구  한성백제역 2번 출구  2020  11   4   0.666667  0.433333  0.999644  14.154084   
125  송파구  한성백제역 2번 출구  2020  11   5   0.800000  0.333333  1.000000  15.285720   
126  송파구  한성백제역 2번 출구  2020  11   6   0.900000  0.433333  0.999996  15.071497   
128  송파구  한성백제역 2번 출구  2020  11   8   6.966667  4.466667  0.999707  14.222225   
129  송파구  한성백제역 2번 출구  2020  11   9   3.500000  2.500000  0.998684  13.555937   
149  송파구  한성백제역 2번 출구  2020  12   5   0.566667  0.200000  1.000000  15.454546   
152  송파구  한성백제역 2번 출구  2020  12   8   4.633333  2.366667  0.999989  14.956069   
153  송파구  한성백제역 2번 출구  2020  12   9   2.100000  1.666667  0.994784  12.494863   
161  송파구  한성백제역 2번 출구  2020  12  17   5.200000  4.066667  0.995667  12.676096   
163  송파구  한성백제역 2번 출구  2020  12  19   3.033333  2.266667  0.997594  13.164392   
164  송파구  한성백제역 2번 출구  2020  12  20   3.033333  1.566667  0.999988  14.932043   
167  송파구  한성백제역 2번 출구  2020  12  23   1.866667  1.133333  0.999866  14.458065   
168  송파구  한성백제역 2번 출구  2021   1   0   0.700000  0.566667  0.993337  12.231370   
169  송파구  한성백제역 2번 출구  2021   1   1   0.433333  0.300000  0.999141  13.782838   
171  송파구  한성백제역 2번 출구  2021   1   3   0.300000  0.233333  0.995958  12.740488   
173  송파구  한성백제역 2번 출구  2021   1   5   0.266667  0.100000  1.000000  15.400001   
174  송파구  한성백제역 2번 출구  2021   1   6   0.900000  0.233333  1.000000  15.650000   
176  송파구  한성백제역 2번 출구  2021   1   8   2.633333  1.600000  0.999865  14.455177   
187  송파구  한성백제역 2번 출구  2021   1  19   1.933333  1.533333  0.994829  12.503628   
188  송파구  한성백제역 2번 출구  2021   1  20   1.800000  1.200000  0.999492  14.017272   
191  송파구  한성백제역 2번 출구  2021   1  23   0.566667  0.366667  0.999667  14.177060   
193  송파구  한성백제역 2번 출구  2021   2   1   0.433333  0.333333  0.996491  12.865487   
195  송파구  한성백제역 2번 출구  2021   2   3   0.333333  0.033333  1.000000  15.888889   
197  송파구  한성백제역 2번 출구  2021   2   5   0.833333  0.066667  1.000000  15.913043   
198  송파구  한성백제역 2번 출구  2021   2   6   1.100000  0.633333  0.999938  14.644285   
200  송파구  한성백제역 2번 출구  2021   2   8   3.400000  1.600000  0.999997  15.111157   
206  송파구  한성백제역 2번 출구  2021   2  14   6.800000  5.300000  0.995849  12.716018   
213  송파구  한성백제역 2번 출구  2021   2  21   3.866667  2.766667  0.998653  13.542453   
215  송파구  한성백제역 2번 출구  2021   2  23   1.300000  0.833333  0.999708  14.223148   
217  송파구  한성백제역 2번 출구  2021   3   1   1.233333  0.900000  0.998244  13.380586   
218  송파구  한성백제역 2번 출구  2021   3   2   0.633333  0.333333  0.999984  14.889199   
219  송파구  한성백제역 2번 출구  2021   3   3 

In [11]:
for i in range(len(df_new)):
    if (df_new['L_s'][i] > df_new['3사분위수'][i]) & (df_new['L_s'][i] > (15*0.7)):
        df_new['내 균형'][i] = '불균형'

# L_s가 Q1~Q3사이이더라도 이때 p_1+가 너무 작으면(50%이하) 불균형
- 따릉이 목표 거치율이 60~70%이므로 한대라도 있을 확률인 p_1+도 유사하게 잡기 위해

In [12]:
t = df_new[(df_new['3사분위수']>=df_new['L_s']) & (df_new['L_s'] >= df_new['1사분위수'])]
t

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
9    송파구  한성백제역 2번 출구  2020   6   9   4.700000   5.200000  0.882838   
10   송파구  한성백제역 2번 출구  2020   6  10   4.666667   4.433333  0.962181   
11   송파구  한성백제역 2번 출구  2020   6  11   5.100000   4.566667  0.978915   
12   송파구  한성백제역 2번 출구  2020   6  12   5.766667   5.266667  0.974154   
13   송파구  한성백제역 2번 출구  2020   6  13   6.266667   5.800000  0.970494   
15   송파구  한성백제역 2번 출구  2020   6  15   8.266667   8.766667  0.909685   
17   송파구  한성백제역 2번 출구  2020   6  17  12.166667  12.866667  0.911342   
18   송파구  한성백제역 2번 출구  2020   6  18  16.566667  16.700000  0.937329   
19   송파구  한성백제역 2번 출구  2020   6  19  15.600000  15.533333  0.943170   
20   송파구  한성백제역 2번 출구  2020   6  20  16.700000  16.533333  0.945778   
21   송파구  한성백제역 2번 출구  2020   6  21  14.700000  15.766667  0.902803   
22   송파구  한성백제역 2번 출구  2020   6  22  14.066667  12.333333  0.983172   
23   송파구  한성백제역 2번 출구  2020   6  23   7.400000   7.933333  0.903085   
24   송파구  한성백제역 2번 출구  2020   7   0   4.266667   3.933333  0.971622   
28   송파구  한성백제역 2번 출구  2020   7   4   0.833333   0.800000  0.958402   
34   송파구  한성백제역 2번 출구  2020   7  10   4.900000   4.733333  0.956036   
36   송파구  한성백제역 2번 출구  2020   7  12   5.300000   4.900000  0.970807   
37   송파구  한성백제역 2번 출구  2020   7  13   5.500000   4.533333  0.991714   
38   송파구  한성백제역 2번 출구  2020   7  14   6.233333   6.333333  0.933392   
39   송파구  한성백제역 2번 출구  2020   7  15   8.066667   7.400000  0.972976   
40   송파구  한성백제역 2번 출구  2020   7  16   9.066667  10.033333  0.882696   
41   송파구  한성백제역 2번 출구  2020   7  17  11.566667  12.133333  0.916079   
42   송파구  한성백제역 2번 출구  2020   7  18  13.333333  14.733333  0.883670   
43   송파구  한성백제역 2번 출구  2020   7  19  13.866667  13.133333  0.963230   
44   송파구  한성백제역 2번 출구  2020   7  20  15.133333  15.233333  0.938026   
45   송파구  한성백제역 2번 출구  2020   7  21  14.500000  14.600000  0.937887   
46   송파구  한성백제역 2번 출구  2020   7  22  11.300000  11.400000  0.936939   
47   송파구  한성백제역 2번 출구  2020   7  23   6.733333   5.733333  0.987872   
48   송파구  한성백제역 2번 출구  2020   8   0   2.433333   2.333333  0.958827   
49   송파구  한성백제역 2번 출구  2020   8   1   1.166667   1.100000  0.964745   
50   송파구  한성백제역 2번 출구  2020   8   2   1.700000   1.566667  0.971715   
52   송파구  한성백제역 2번 출구  2020   8   4   0.800000   0.666667  0.990560   
54   송파구  한성백제역 2번 출구  2020   8   6   0.733333   0.700000  0.960490   
56   송파구  한성백제역 2번 출구  2020   8   8   3.933333   3.300000  0.989780   
57   송파구  한성백제역 2번 출구  2020   8   9   2.533333   2.533333  0.941176   
58   송파구  한성백제역 2번 출구  2020   8  10   3.033333   3.100000  0.930398   
60   송파구  한성백제역 2번 출구  2020   8  12   3.500000   3.366667  0.957658   
61   송파구  한성백제역 2번 출구  2020   8  13   4.333333   4.366667  0.937502   
64   송파구  한성백제역 2번 출구  2020   8  16   5.100000   4.400000  0.985924   
65   송파구  한성백제역 2번 출구  2020   8  17   6.566667   6.866667  0.917888   
67   송파구  한성백제역 2번 출구  2020   8  19   8.800000   7.866667  0.979280   
68   송파구  한성백제역 2번 출구  2020   8  20   8.366667   8.333333  0.943036   
69   송파구  한성백제역 2번 출구  2020   8  21   7.533333   7.533333  0.941176   
70   송파구  한성백제역 2번 출구  2020   8  22   5.500000   5.900000  0.902708   
71   송파구  한성백제역 2번 출구  2020   8  23   3.833333   3.166667  0.991489   
73   송파구  한성백제역 2번 출구  2020   9   1   2.066667   2.200000  0.907405   
74   송파구  한성백제역 2번 출구  2020   9   2   1.566667   1.433333  0.973695   
76   송파구  한성백제역 2번 출구  2020   9   4   0.400000   0.333333  0.990560   
78   송파구  한성백제역 2번 출구  2020   9   6   1.000000   1.000000  0.941176   
80   송파구  한성백제역 2번 출구  2020   9   8   6.233333   5.233333  0.989696   
81   송파구  한성백제역 2번 출구  2020   9   9   3.733333   3.200000  0.986921   
82   송파구  한성백제역 2번 출구  2020   9  10   4.466667   3.900000  0.983922   
83   송파구  한성백제역 2번 출구  2020   9  11   4.766667   5.200000  0.892080   
84   송파구  한성백제역 2번 출구  2020   9  12   6.133333   6.166667  0.938591   
85   송파구  한성백제역 2번 출구  2020   9  13   7.200000   6.966667  0.955388   
86   송파구  한성백제역 2번 출구  2020  

In [13]:
t[t['p_1+']< 0.6]

Empty DataFrame
Columns: [자치구, 대여소명, 년, 월, 시간, lambda, mu, p_1+, L_s, W_s, 재고량, 3사분위수, 2사분위수, 1사분위수, 균형, 내 균형]
Index: []

- 위 결과를 보면 l_s가 q1~q3사이에 있지만 자전거가 한대라도 있을 확률이 0.5이하이므로 불균형이다
- 특히 19시, 20시의 경우 퇴근시간이므로 이시간에 불균형이면 이용에 불편을 초래함
- 또한 21년 1월 19시의 경우 불균형인데도 불구하고 기존연구방법에서는 이를 균형으로 판단

In [14]:
for i in range(len(df_new)):
    if (df_new['3사분위수'][i]>=df_new['L_s'][i]) & (df_new['L_s'][i] >= df_new['1사분위수'][i]) & (df_new['p_1+'][i] < 0.6):
        df_new['내 균형'][i] = '불균형'

In [15]:
df_new

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  한성백제역 2번 출구  2020   6   0   4.500000   3.066667  0.999310   
1    송파구  한성백제역 2번 출구  2020   6   1   2.966667   2.133333  0.998559   
2    송파구  한성백제역 2번 출구  2020   6   2   1.800000   2.100000  0.845933   
3    송파구  한성백제역 2번 출구  2020   6   3   1.233333   1.833333  0.672339   
4    송파구  한성백제역 2번 출구  2020   6   4   0.800000   0.566667  0.998825   
5    송파구  한성백제역 2번 출구  2020   6   5   1.000000   0.700000  0.999001   
6    송파구  한성백제역 2번 출구  2020   6   6   2.200000   1.800000  0.992418   
7    송파구  한성백제역 2번 출구  2020   6   7   2.400000   4.700000  0.510633   
8    송파구  한성백제역 2번 출구  2020   6   8   8.766667   6.633333  0.997166   
9    송파구  한성백제역 2번 출구  2020   6   9   4.700000   5.200000  0.882838   
10   송파구  한성백제역 2번 출구  2020   6  10   4.666667   4.433333  0.962181   
11   송파구  한성백제역 2번 출구  2020   6  11   5.100000   4.566667  0.978915   
12   송파구  한성백제역 2번 출구  2020   6  12   5.766667   5.266667  0.974154   
13   송파구  한성백제역 2번 출구  2020   6  13   6.266667   5.800000  0.970494   
14   송파구  한성백제역 2번 출구  2020   6  14   5.800000   6.700000  0.853016   
15   송파구  한성백제역 2번 출구  2020   6  15   8.266667   8.766667  0.909685   
16   송파구  한성백제역 2번 출구  2020   6  16   8.900000  10.866667  0.812731   
17   송파구  한성백제역 2번 출구  2020   6  17  12.166667  12.866667  0.911342   
18   송파구  한성백제역 2번 출구  2020   6  18  16.566667  16.700000  0.937329   
19   송파구  한성백제역 2번 출구  2020   6  19  15.600000  15.533333  0.943170   
20   송파구  한성백제역 2번 출구  2020   6  20  16.700000  16.533333  0.945778   
21   송파구  한성백제역 2번 출구  2020   6  21  14.700000  15.766667  0.902803   
22   송파구  한성백제역 2번 출구  2020   6  22  14.066667  12.333333  0.983172   
23   송파구  한성백제역 2번 출구  2020   6  23   7.400000   7.933333  0.903085   
24   송파구  한성백제역 2번 출구  2020   7   0   4.266667   3.933333  0.971622   
25   송파구  한성백제역 2번 출구  2020   7   1   3.300000   2.400000  0.998322   
26   송파구  한성백제역 2번 출구  2020   7   2   1.933333   2.333333  0.821263   
27   송파구  한성백제역 2번 출구  2020   7   3   1.133333   1.700000  0.666328   
28   송파구  한성백제역 2번 출구  2020   7   4   0.833333   0.800000  0.958402   
29   송파구  한성백제역 2번 출구  2020   7   5   1.066667   0.600000  0.999956   
30   송파구  한성백제역 2번 출구  2020   7   6   1.233333   1.533333  0.799395   
31   송파구  한성백제역 2번 출구  2020   7   7   1.100000   2.300000  0.478259   
32   송파구  한성백제역 2번 출구  2020   7   8   6.300000   4.566667  0.998395   
33   송파구  한성백제역 2번 출구  2020   7   9   4.600000   3.233333  0.998943   
34   송파구  한성백제역 2번 출구  2020   7  10   4.900000   4.733333  0.956036   
35   송파구  한성백제역 2번 출구  2020   7  11   3.733333   5.033333  0.740104   
36   송파구  한성백제역 2번 출구  2020   7  12   5.300000   4.900000  0.970807   
37   송파구  한성백제역 2번 출구  2020   7  13   5.500000   4.533333  0.991714   
38   송파구  한성백제역 2번 출구  2020   7  14   6.233333   6.333333  0.933392   
39   송파구  한성백제역 2번 출구  2020   7  15   8.066667   7.400000  0.972976   
40   송파구  한성백제역 2번 출구  2020   7  16   9.066667  10.033333  0.882696   
41   송파구  한성백제역 2번 출구  2020   7  17  11.566667  12.133333  0.916079   
42   송파구  한성백제역 2번 출구  2020   7  18  13.333333  14.733333  0.883670   
43   송파구  한성백제역 2번 출구  2020   7  19  13.866667  13.133333  0.963230   
44   송파구  한성백제역 2번 출구  2020   7  20  15.133333  15.233333  0.938026   
45   송파구  한성백제역 2번 출구  2020   7  21  14.500000  14.600000  0.937887   
46   송파구  한성백제역 2번 출구  2020   7  22  11.300000  11.400000  0.936939   
47   송파구  한성백제역 2번 출구  2020   7  23   6.733333   5.733333  0.987872   
48   송파구  한성백제역 2번 출구  2020   8   0   2.433333   2.333333  0.958827   
49   송파구  한성백제역 2번 출구  2020   8   1   1.166667   1.100000  0.964745   
50   송파구  한성백제역 2번 출구  2020   8   2   1.700000   1.566667  0.971715   
51   송파구  한성백제역 2번 출구  2020   8   3   1.300000   1.733333  0.748107   
52   송파구  한성백제역 2번 출구  2020   8   4   0.800000   0.666667  0.990560   
53   송파구  한성백제역 2번 출구  2020   8   5   0.800000   0.566667  0.998825   
54   송파구  한성백제역 2번 출구  2020   8   6   0.733333   0.700000  0.960490   
55   송파구  한성백제역 2번 출구  2020  

In [16]:
len(df_new[df_new['균형'] == '불균형'])

142

In [17]:
len(df_new[df_new['내 균형'] == '불균형'])

76

In [18]:
len(df_new)

288

In [19]:
# TN

len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '불균형')])

43

In [20]:
# TP

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')])

113

In [21]:
# FN

len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')])

99

In [22]:
# FP

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')])

33

In [23]:
# 정확도(기존방법과 내 방법의 일치율)

(len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '불균형')])) / len(df_new)

0.5416666666666666

In [24]:
# 정밀도(기존 방법에서 균형으로 판단한 데이터 중 내가 균형이라고 판단한 비율)

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) / (len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')]))

0.773972602739726

In [25]:
# 재현율(내 방법에서 균형으로 판단한 것 중 기존 방법에서도 균형으로 판단한 비율)

len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) / (len(df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '균형')]) + len(df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')]))

0.5330188679245284

In [26]:
a = df_new[(df_new['균형'] == '불균형') & (df_new['내 균형'] == '균형')]
a

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
2    송파구  한성백제역 2번 출구  2020   6   2   1.800000   2.100000  0.845933   
3    송파구  한성백제역 2번 출구  2020   6   3   1.233333   1.833333  0.672339   
9    송파구  한성백제역 2번 출구  2020   6   9   4.700000   5.200000  0.882838   
10   송파구  한성백제역 2번 출구  2020   6  10   4.666667   4.433333  0.962181   
15   송파구  한성백제역 2번 출구  2020   6  15   8.266667   8.766667  0.909685   
16   송파구  한성백제역 2번 출구  2020   6  16   8.900000  10.866667  0.812731   
17   송파구  한성백제역 2번 출구  2020   6  17  12.166667  12.866667  0.911342   
18   송파구  한성백제역 2번 출구  2020   6  18  16.566667  16.700000  0.937329   
19   송파구  한성백제역 2번 출구  2020   6  19  15.600000  15.533333  0.943170   
20   송파구  한성백제역 2번 출구  2020   6  20  16.700000  16.533333  0.945778   
21   송파구  한성백제역 2번 출구  2020   6  21  14.700000  15.766667  0.902803   
22   송파구  한성백제역 2번 출구  2020   6  22  14.066667  12.333333  0.983172   
23   송파구  한성백제역 2번 출구  2020   6  23   7.400000   7.933333  0.903085   
34   송파구  한성백제역 2번 출구  2020   7  10   4.900000   4.733333  0.956036   
35   송파구  한성백제역 2번 출구  2020   7  11   3.733333   5.033333  0.740104   
36   송파구  한성백제역 2번 출구  2020   7  12   5.300000   4.900000  0.970807   
37   송파구  한성백제역 2번 출구  2020   7  13   5.500000   4.533333  0.991714   
38   송파구  한성백제역 2번 출구  2020   7  14   6.233333   6.333333  0.933392   
40   송파구  한성백제역 2번 출구  2020   7  16   9.066667  10.033333  0.882696   
41   송파구  한성백제역 2번 출구  2020   7  17  11.566667  12.133333  0.916079   
42   송파구  한성백제역 2번 출구  2020   7  18  13.333333  14.733333  0.883670   
43   송파구  한성백제역 2번 출구  2020   7  19  13.866667  13.133333  0.963230   
44   송파구  한성백제역 2번 출구  2020   7  20  15.133333  15.233333  0.938026   
45   송파구  한성백제역 2번 출구  2020   7  21  14.500000  14.600000  0.937887   
46   송파구  한성백제역 2번 출구  2020   7  22  11.300000  11.400000  0.936939   
47   송파구  한성백제역 2번 출구  2020   7  23   6.733333   5.733333  0.987872   
56   송파구  한성백제역 2번 출구  2020   8   8   3.933333   3.300000  0.989780   
57   송파구  한성백제역 2번 출구  2020   8   9   2.533333   2.533333  0.941176   
58   송파구  한성백제역 2번 출구  2020   8  10   3.033333   3.100000  0.930398   
60   송파구  한성백제역 2번 출구  2020   8  12   3.500000   3.366667  0.957658   
62   송파구  한성백제역 2번 출구  2020   8  14   2.833333   3.266667  0.854391   
63   송파구  한성백제역 2번 출구  2020   8  15   3.400000   3.900000  0.858014   
65   송파구  한성백제역 2번 출구  2020   8  17   6.566667   6.866667  0.917888   
66   송파구  한성백제역 2번 출구  2020   8  18   6.833333   7.766667  0.864452   
67   송파구  한성백제역 2번 출구  2020   8  19   8.800000   7.866667  0.979280   
68   송파구  한성백제역 2번 출구  2020   8  20   8.366667   8.333333  0.943036   
69   송파구  한성백제역 2번 출구  2020   8  21   7.533333   7.533333  0.941176   
70   송파구  한성백제역 2번 출구  2020   8  22   5.500000   5.900000  0.902708   
102  송파구  한성백제역 2번 출구  2020  10   6   0.966667   0.800000  0.991303   
109  송파구  한성백제역 2번 출구  2020  10  13   7.900000   8.733333  0.883378   
122  송파구  한성백제역 2번 출구  2020  11   2   1.866667   1.633333  0.983541   
123  송파구  한성백제역 2번 출구  2020  11   3   1.333333   1.100000  0.991622   
154  송파구  한성백제역 2번 출구  2020  12  10   2.200000   2.300000  0.918013   
155  송파구  한성백제역 2번 출구  2020  12  11   2.500000   2.533333  0.934739   
156  송파구  한성백제역 2번 출구  2020  12  12   2.766667   2.733333  0.946707   
157  송파구  한성백제역 2번 출구  2020  12  13   3.166667   3.700000  0.844852   
158  송파구  한성백제역 2번 출구  2020  12  14   5.266667   4.766667  0.976432   
159  송파구  한성백제역 2번 출구  2020  12  15   4.133333   4.500000  0.893380   
160  송파구  한성백제역 2번 출구  2020  12  16   4.000000   4.366667  0.891632   
162  송파구  한성백제역 2번 출구  2020  12  18   3.833333   5.000000  0.764090   
165  송파구  한성백제역 2번 출구  2020  12  21   2.100000   3.066667  0.684277   
166  송파구  한성백제역 2번 출구  2020  12  22   1.766667   1.566667  0.980973   
177  송파구  한성백제역 2번 출구  2021   1   9   1.233333   1.166667  0.963650   
178  송파구  한성백제역 2번 출구  2021   1  10   1.000000   0.900000  0.977761   
180  송파구  한성백제역 2번 출구  2021   1  12   1.433333   1.800000  0.791967   
181  송파구  한성백제역 2번 출구  2021  

In [27]:
len(a)

99

In [28]:
len(a[a['p_1+'] > 0.9])

64

In [29]:
len(a[a['p_1+'] > 0.8])

87

In [30]:
len(a[a['p_1+'] > 0.7])

94

In [31]:
len(a[a['p_1+'] > 0.6])

99

- 기존방법이 불균형으로 판단한 데이터 중 본 연구에서 균형으로 판단한 데이터는 99개이지만
- 이 중 대여소에 적어도 자전거가 1대 있을 확률이 70%가 넘는 데이터가 94개로 무려 94%이상의 경우에 이용자가 대여소에 도착했을 때 자전거를 빌릴 수 있는 확률이 70%이상이었다.
- 따라서 이러한 경우에는 자전거 재배치를 위한 대여소 내 재고의 균형/ 불균형을 판단할 경우에는 재배치를 할 필요가 없다고 판단하여 균형으로 판단할 수 있다.

In [32]:
b = df_new[(df_new['균형'] == '균형') & (df_new['내 균형'] == '불균형')]
b

자치구         대여소명     년   월  시간     lambda        mu      p_1+        L_s  \
6    송파구  한성백제역 2번 출구  2020   6   6   2.200000  1.800000  0.992418  12.080052   
25   송파구  한성백제역 2번 출구  2020   7   1   3.300000  2.400000  0.998322  13.409408   
29   송파구  한성백제역 2번 출구  2020   7   5   1.066667  0.600000  0.999956  14.715246   
31   송파구  한성백제역 2번 출구  2020   7   7   1.100000  2.300000  0.478259   0.916606   
53   송파구  한성백제역 2번 출구  2020   8   5   0.800000  0.566667  0.998825  13.619927   
55   송파구  한성백제역 2번 출구  2020   8   7   0.666667  1.400000  0.476189   0.909034   
77   송파구  한성백제역 2번 출구  2020   9   5   0.800000  0.200000  1.000000  15.666667   
79   송파구  한성백제역 2번 출구  2020   9   7   1.900000  3.633333  0.522928   1.095876   
95   송파구  한성백제역 2번 출구  2020   9  23   6.466667  5.133333  0.994771  12.492238   
98   송파구  한성백제역 2번 출구  2020  10   2   1.966667  1.300000  0.999549  14.064943   
128  송파구  한성백제역 2번 출구  2020  11   8   6.966667  4.466667  0.999707  14.222225   
129  송파구  한성백제역 2번 출구  2020  11   9   3.500000  2.500000  0.998684  13.555937   
146  송파구  한성백제역 2번 출구  2020  12   2   0.300000  1.066667  0.281250   0.391304   
149  송파구  한성백제역 2번 출구  2020  12   5   0.566667  0.200000  1.000000  15.454546   
167  송파구  한성백제역 2번 출구  2020  12  23   1.866667  1.133333  0.999866  14.458065   
168  송파구  한성백제역 2번 출구  2021   1   0   0.700000  0.566667  0.993337  12.231370   
169  송파구  한성백제역 2번 출구  2021   1   1   0.433333  0.300000  0.999141  13.782838   
170  송파구  한성백제역 2번 출구  2021   1   2   0.100000  0.200000  0.499996   0.999870   
171  송파구  한성백제역 2번 출구  2021   1   3   0.300000  0.233333  0.995958  12.740488   
172  송파구  한성백제역 2번 출구  2021   1   4   0.000000  0.000000  0.000000   0.000000   
173  송파구  한성백제역 2번 출구  2021   1   5   0.266667  0.100000  1.000000  15.400001   
174  송파구  한성백제역 2번 출구  2021   1   6   0.900000  0.233333  1.000000  15.650000   
187  송파구  한성백제역 2번 출구  2021   1  19   1.933333  1.533333  0.994829  12.503628   
188  송파구  한성백제역 2번 출구  2021   1  20   1.800000  1.200000  0.999492  14.017272   
191  송파구  한성백제역 2번 출구  2021   1  23   0.566667  0.366667  0.999667  14.177060   
241  송파구  한성백제역 2번 출구  2021   4   1   1.266667  0.900000  0.998775  13.596577   
242  송파구  한성백제역 2번 출구  2021   4   2   0.933333  0.633333  0.999350  13.912233   
243  송파구  한성백제역 2번 출구  2021   4   3   0.200000  0.400000  0.499996   0.999870   
245  송파구  한성백제역 2번 출구  2021   4   5   0.633333  0.233333  1.000000  15.416667   
246  송파구  한성백제역 2번 출구  2021   4   6   0.600000  1.366667  0.439024   0.782594   
247  송파구  한성백제역 2번 출구  2021   4   7   1.866667  3.366667  0.554436   1.243692   
248  송파구  한성백제역 2번 출구  2021   4   8  11.800000  6.766667  0.999942  14.656962   
263  송파구  한성백제역 2번 출구  2021   4  23   3.866667  2.933333  0.997069  13.013764   

            W_s        재고량      3사분위수     2사분위수     1사분위수  균형 내 균형  
6      6.762415   5.933333  12.066667  7.633333  5.566667  균형  불균형  
25     5.596645   6.666667  12.066667  7.633333  5.566667  균형  불균형  
29    24.526489   7.466667  12.066667  7.633333  5.566667  균형  불균형  
31     0.833281   6.633333  12.066667  7.633333  5.566667  균형  불균형  
53    24.063432   6.866667  12.066667  7.633333  5.566667  균형  불균형  
55     1.363556   6.766667  12.066667  7.633333  5.566667  균형  불균형  
77    78.333333  10.666667  12.066667  7.633333  5.566667  균형  불균형  
79     0.576785   9.666667  12.066667  7.633333  5.566667  균형  불균형  
95     2.446345   9.966667  12.066667  7.633333  5.566667  균형  불균형  
98    10.824066  11.700000  12.066667  7.633333  5.566667  균형  불균형  
128    3.185013   8.366667  12.066667  7.633333  5.566667  균형  불균형  
129    5.429521   9.366667  12.066667  7.633333  5.566667  균형  불균형  
146    1.304348   6.766667  12.066667  7.633333  5.566667  균형  불균형  
149   77.272732   7.433333  12.066667  7.633333  5.566667  균형  불균형  
167   12.758825   6.333333  12.066667  7.633333  5.566667  균형  불균형  
168   21.729545   8.866667  12.066667  7.633333  5.566667  균형  불균형  
169   45.982268   9.200000  12.066667  7.633333  5.566667  균형  불균형  
170    9

In [33]:
len(b)

33

In [34]:
len(b[(b['lambda'] > b['mu']) & (b['L_s'] > (15*0.7))])

24

In [35]:
24/33

0.7272727272727273

- 기존방법이 균형으로 판단한 데이터 중 본 연구에서 불균형으로 판단한 데이터는 33개이지만
- 이 중 반납이 대여보다 많고 대여소 내 평균 자전거수(L_s)가 서울시 거치대율의 목표인 거치대수의 60~70%를 초과하는 경우가 24개로 72%이상의 경우에 불필요하게 자전거의 재고가 많다고 판단된다.
- 따라서 이러한 경우에는 이용자들의 자전거 이용에는 문제가 없지만 주변 대여소의 재고가 부족하다면 자전거 재배치의 측면에서는 재배치를 할 필요가 있다고 판단하여 불균형으로 판단할 수 있다.

In [36]:
df_new

자치구         대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  한성백제역 2번 출구  2020   6   0   4.500000   3.066667  0.999310   
1    송파구  한성백제역 2번 출구  2020   6   1   2.966667   2.133333  0.998559   
2    송파구  한성백제역 2번 출구  2020   6   2   1.800000   2.100000  0.845933   
3    송파구  한성백제역 2번 출구  2020   6   3   1.233333   1.833333  0.672339   
4    송파구  한성백제역 2번 출구  2020   6   4   0.800000   0.566667  0.998825   
5    송파구  한성백제역 2번 출구  2020   6   5   1.000000   0.700000  0.999001   
6    송파구  한성백제역 2번 출구  2020   6   6   2.200000   1.800000  0.992418   
7    송파구  한성백제역 2번 출구  2020   6   7   2.400000   4.700000  0.510633   
8    송파구  한성백제역 2번 출구  2020   6   8   8.766667   6.633333  0.997166   
9    송파구  한성백제역 2번 출구  2020   6   9   4.700000   5.200000  0.882838   
10   송파구  한성백제역 2번 출구  2020   6  10   4.666667   4.433333  0.962181   
11   송파구  한성백제역 2번 출구  2020   6  11   5.100000   4.566667  0.978915   
12   송파구  한성백제역 2번 출구  2020   6  12   5.766667   5.266667  0.974154   
13   송파구  한성백제역 2번 출구  2020   6  13   6.266667   5.800000  0.970494   
14   송파구  한성백제역 2번 출구  2020   6  14   5.800000   6.700000  0.853016   
15   송파구  한성백제역 2번 출구  2020   6  15   8.266667   8.766667  0.909685   
16   송파구  한성백제역 2번 출구  2020   6  16   8.900000  10.866667  0.812731   
17   송파구  한성백제역 2번 출구  2020   6  17  12.166667  12.866667  0.911342   
18   송파구  한성백제역 2번 출구  2020   6  18  16.566667  16.700000  0.937329   
19   송파구  한성백제역 2번 출구  2020   6  19  15.600000  15.533333  0.943170   
20   송파구  한성백제역 2번 출구  2020   6  20  16.700000  16.533333  0.945778   
21   송파구  한성백제역 2번 출구  2020   6  21  14.700000  15.766667  0.902803   
22   송파구  한성백제역 2번 출구  2020   6  22  14.066667  12.333333  0.983172   
23   송파구  한성백제역 2번 출구  2020   6  23   7.400000   7.933333  0.903085   
24   송파구  한성백제역 2번 출구  2020   7   0   4.266667   3.933333  0.971622   
25   송파구  한성백제역 2번 출구  2020   7   1   3.300000   2.400000  0.998322   
26   송파구  한성백제역 2번 출구  2020   7   2   1.933333   2.333333  0.821263   
27   송파구  한성백제역 2번 출구  2020   7   3   1.133333   1.700000  0.666328   
28   송파구  한성백제역 2번 출구  2020   7   4   0.833333   0.800000  0.958402   
29   송파구  한성백제역 2번 출구  2020   7   5   1.066667   0.600000  0.999956   
30   송파구  한성백제역 2번 출구  2020   7   6   1.233333   1.533333  0.799395   
31   송파구  한성백제역 2번 출구  2020   7   7   1.100000   2.300000  0.478259   
32   송파구  한성백제역 2번 출구  2020   7   8   6.300000   4.566667  0.998395   
33   송파구  한성백제역 2번 출구  2020   7   9   4.600000   3.233333  0.998943   
34   송파구  한성백제역 2번 출구  2020   7  10   4.900000   4.733333  0.956036   
35   송파구  한성백제역 2번 출구  2020   7  11   3.733333   5.033333  0.740104   
36   송파구  한성백제역 2번 출구  2020   7  12   5.300000   4.900000  0.970807   
37   송파구  한성백제역 2번 출구  2020   7  13   5.500000   4.533333  0.991714   
38   송파구  한성백제역 2번 출구  2020   7  14   6.233333   6.333333  0.933392   
39   송파구  한성백제역 2번 출구  2020   7  15   8.066667   7.400000  0.972976   
40   송파구  한성백제역 2번 출구  2020   7  16   9.066667  10.033333  0.882696   
41   송파구  한성백제역 2번 출구  2020   7  17  11.566667  12.133333  0.916079   
42   송파구  한성백제역 2번 출구  2020   7  18  13.333333  14.733333  0.883670   
43   송파구  한성백제역 2번 출구  2020   7  19  13.866667  13.133333  0.963230   
44   송파구  한성백제역 2번 출구  2020   7  20  15.133333  15.233333  0.938026   
45   송파구  한성백제역 2번 출구  2020   7  21  14.500000  14.600000  0.937887   
46   송파구  한성백제역 2번 출구  2020   7  22  11.300000  11.400000  0.936939   
47   송파구  한성백제역 2번 출구  2020   7  23   6.733333   5.733333  0.987872   
48   송파구  한성백제역 2번 출구  2020   8   0   2.433333   2.333333  0.958827   
49   송파구  한성백제역 2번 출구  2020   8   1   1.166667   1.100000  0.964745   
50   송파구  한성백제역 2번 출구  2020   8   2   1.700000   1.566667  0.971715   
51   송파구  한성백제역 2번 출구  2020   8   3   1.300000   1.733333  0.748107   
52   송파구  한성백제역 2번 출구  2020   8   4   0.800000   0.666667  0.990560   
53   송파구  한성백제역 2번 출구  2020   8   5   0.800000   0.566667  0.998825   
54   송파구  한성백제역 2번 출구  2020   8   6   0.733333   0.700000  0.960490   
55   송파구  한성백제역 2번 출구  2020  

In [37]:
df_new.to_csv('./한성백제역 2번 출구.csv', index = False, encoding = 'cp949')

# confusion matrix

In [23]:
len(df_new)

288

In [24]:
# 재고량이 ls보다 많거나 같을 경우 불균형 아님 = 균형임

z = df_new[df_new['L_s'] <= df_new['재고량']]
z

자치구               대여소명     년   월  시간     lambda         mu      p_1+  \
1    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   1   1.033333   1.300000  0.794608   
2    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   2   0.500000   0.666667  0.749940   
11   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  11  10.333333   8.333333  0.999530   
12   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  12  12.266667   8.766667  0.999977   
13   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  13  11.433333  12.033333  0.935505   
14   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  14  13.566667  15.266667  0.884895   
15   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  15  12.866667  17.133333  0.750911   
16   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  16  16.500000  20.166667  0.817640   
17   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  17  19.266667  28.900000  0.666664   
35   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  11   7.533333   7.000000  0.989718   
37   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  13  11.233333  10.133333  0.994244   
38   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  14  11.000000  15.633333  0.703614   
39   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  15  12.700000  15.433333  0.822271   
40   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  16  13.166667  19.166667  0.686951   
41   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  17  17.966667  23.166667  0.775398   
50   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   2   0.233333   0.433333  0.538462   
51   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   3   0.200000   0.233333  0.855489   
56   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   8   6.666667   8.733333  0.763265   
62   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  14   5.566667   7.033333  0.791232   
63   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  15   6.466667   7.500000  0.860325   
64   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  16   6.433333   8.366667  0.768811   
65   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  17   9.566667  11.500000  0.831072   
74   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9   2   0.233333   0.433333  0.538462   
85   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  13  14.566667  14.300000  0.973692   
86   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  14  14.100000  18.833333  0.748616   
87   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  15  16.600000  18.833333  0.878286   
88   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  16  18.200000  21.666667  0.838974   
89   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9  17  19.700000  25.366667  0.776463   
108  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  12  14.633333  13.066667  0.995331   
109  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  13  15.866667  16.700000  0.935477   
110  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  14  16.866667  21.466667  0.785517   
111  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  15  18.700000  19.600000  0.938294   
112  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  10  16  22.033333  26.733333  0.823542   
120  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   0   0.700000   0.500000  0.999977   
121  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   1   1.100000   0.833333  0.999898   
122  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   2   0.466667   0.466667  0.965517   
123  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   3   0.033333   0.233333  0.142857   
124  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   4   0.266667   0.300000  0.885115   
126  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   6   1.966667   0.333333  1.000000   
127  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   7   8.833333   3.500000  1.000000   
128  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   8  16.366667   9.500000  1.000000   
129  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11   9  10.900000   6.900000  0.999999   
130  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  10   8.433333   4.100000  1.000000   
131  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  11   6.733333   4.966667  0.999948   
132  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  12   9.200000   8.600000  0.988505   
133  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  13   9.800000   9.633333  0.973156   
134  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  14  11.066667  11.700000  0.932411   
135  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  15  13.500000  12.000000  0.995754   
136  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  16  13.666667  17.500000  0.780784   
137  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  17  14.700000  21.833333  0.673279   
138  송파구  롯데월드타워(잠실역2번출구 쪽)  2020  11  18  16.400000  25.033333  0.655125   
1

In [25]:
len(z)

89

In [26]:
# TP

len(z[z['균형'] == '균형'])

34

In [27]:
# FN

len(z[z['균형'] == '불균형'])

55

- 재고량이 ls보다 많거나 같을 경우 균형인데 불균형이라고 판단한 수가 55개

- 균형인데 균형이라고 판단한 수가 34개

In [28]:
# ls보다 재고량이 작으므로 불균형

w = df_new[df_new['L_s'] > df_new['재고량']]
w

자치구               대여소명     년   월  시간     lambda         mu      p_1+  \
0    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   0   1.666667   1.533333  0.991495   
3    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   3   0.366667   0.333333  0.993272   
4    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   4   0.833333   0.466667  1.000000   
5    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   5   1.200000   0.300000  1.000000   
6    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   6   2.933333   1.266667  1.000000   
7    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   7  10.200000   5.666667  1.000000   
8    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   8  19.033333  17.500000  0.991595   
9    송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6   9  17.633333   9.533333  1.000000   
10   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  10   9.733333   5.800000  1.000000   
18   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  18  27.400000  30.466667  0.894477   
19   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  19  17.966667  18.433333  0.951743   
20   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  20  14.533333  14.766667  0.957284   
21   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  21  11.633333  11.666667  0.964118   
22   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  22   7.933333   7.800000  0.973073   
23   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   6  23   2.966667   3.266667  0.902176   
24   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   0   1.733333   1.900000  0.905701   
25   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   1   0.900000   0.766667  0.998321   
26   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   2   0.433333   0.500000  0.864531   
27   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   3   0.633333   0.400000  0.999999   
28   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   4   0.766667   0.200000  1.000000   
29   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   5   1.266667   0.800000  0.999999   
30   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   6   2.533333   1.566667  0.999999   
31   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   7  10.033333   4.233333  1.000000   
32   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   8  15.533333  13.733333  0.996209   
33   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7   9  13.333333   6.400000  1.000000   
34   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  10   7.800000   5.266667  0.999995   
36   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  12   9.700000   7.600000  0.999766   
42   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  18  21.366667  22.800000  0.925854   
43   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  19  15.400000  15.066667  0.975037   
44   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  20  11.600000  11.700000  0.961214   
45   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  21   8.366667   8.666667  0.945913   
46   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  22   5.700000   6.033333  0.931589   
47   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   7  23   2.800000   2.900000  0.945999   
48   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   0   1.333333   1.233333  0.990563   
49   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   1   0.900000   0.866667  0.980649   
52   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   4   0.166667   0.066667  1.000000   
53   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   5   0.733333   0.300000  1.000000   
54   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   6   1.400000   0.566667  1.000000   
55   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   7   5.866667   3.433333  1.000000   
57   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8   9   7.766667   3.900000  1.000000   
58   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  10   5.433333   4.166667  0.999862   
59   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  11   5.566667   5.100000  0.992158   
60   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  12   5.266667   4.366667  0.999097   
61   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  13   5.933333   5.133333  0.997628   
66   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  18  10.333333  10.700000  0.946138   
67   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  19   8.800000   8.466667  0.980928   
68   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  20   6.733333   7.233333  0.920976   
69   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  21   5.100000   5.633333  0.899721   
70   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  22   3.733333   3.700000  0.969672   
71   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   8  23   1.833333   2.066667  0.883487   
72   송파구  롯데월드타워(잠실역2번출구 쪽)  2020   9   0   2.600000   2.300000  0.996164   
7

In [29]:
len(w)

199

In [30]:
# FP

len(w[w['균형'] == '균형'])

111

In [31]:
# TN

len(w[w['균형'] == '불균형'])

88

- 불균형인데 균형이라고 판단한 수는 111
- 불균형인데 불균형이라고 판단한 수는 88

# confusion matrix

0 |균형|불균형
-|-|-
균형|TP|FP
불균형|FN|TN


1. 정확도 accuracy = 실제(내) 데이터와 예측(기존방법)데이터가 얼마나 같은지를 평가하는 지표 = TP + TN / TP+TN+FP+FN
2. 정밀도 precision = 균형으로 예측한 데이터 중 실제로 균형인 데이터 비율 = TP / TP+FP
3. 재현율 recall = 실제 균형 중 균형으로 예측한 비율 = TP / TP+FN

=> 정밀도에 초점. (불균형을 균형으로 예측하여 재배치하지 않으면 서비스에 차질)


In [32]:
# 정확도

(len(z[z['균형'] == '균형']) + len(w[w['균형'] == '불균형'])) / len(df_new)

0.4236111111111111

In [33]:
# 정밀도

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(w[w['균형'] == '균형']))

0.23448275862068965

In [34]:
# 재현율

len(z[z['균형'] == '균형']) / (len(z[z['균형'] == '균형']) + len(z[z['균형'] == '불균형']))

0.38202247191011235